tests/virt_autotest/sev_snp_guest_verification.pm

```diff
@@ -0,0 +1,1270 @@
+        my $smm_details = script_output("grep -A2 -B2 '<smm' $xml_file || echo 'No SMM details found'", proceed_on_failure => 1);
+        record_info('SMM Incompatibility', "SMM configuration found in VM XML which is incompatible with SEV-SNP: $smm_details", result => 'softfail');
+        script_run("sed -i '/<smm[^>]*>/d' $xml_file");
+        record_info('SMM Removal', 'Removed SMM tags from XML file');
+    }
+
+    # Check for memoryBacking with locked configuration
+    my $memory_backing_count = script_output("grep -c '<memoryBacking>' $xml_file || echo '0'", proceed_on_failure => 1);
+    if ($memory_backing_count ne '0') {
+        my $memory_backing_details = script_output("grep -A3 -B1 '<memoryBacking>' $xml_file || echo 'No memoryBacking details found'", proceed_on_failure => 1);
+        record_info('MemoryBacking Incompatibility', "memoryBacking configuration found in VM XML which is incompatible with SEV-SNP: $memory_backing_details", result => 'softfail');
+        script_run("sed -i '/<memoryBacking>/,/<\\/memoryBacking>/d' $xml_file");
+        record_info('MemoryBacking Removal', 'Removed memoryBacking tags from XML file');
+    }
+
+    # Delete existing OS section and insert SEV-compatible OS configuration
+    record_info('OS Section', "Updating OS section for SEV-SNP compatibility");
+
+    # First delete the existing OS section (if exists)
+    script_run("sed -i '/<os/,/<\\/os>/d' $xml_file");
+
+    # Define new OS section with SEV-compatible settings
+    my $os_section = <<EOL;
+<os>
+  <type arch="x86_64" machine="$machine_type">hvm</type>
+  <loader readonly="yes" type="rom">/usr/share/qemu/ovmf-x86_64-sev.bin</loader>
+  <boot dev="hd"/>
+</os>
+EOL
+
+    # A safer approach: Write OS section to a temporary file and insert it using sed
+    script_run("echo '$os_section' > /tmp/os_section.xml");
+    # Insert the OS section right before the </domain> tag - using sed -e command
+    script_run("sed -i -e '/<\\/domain>/e cat /tmp/os_section.xml' $xml_file");
+
+    # Verify that changes were applied successfully
+    my $new_os_section = script_output("grep -A5 '<os>' $xml_file || echo 'OS section not found'", proceed_on_failure => 1);
+    if ($new_os_section !~ /ovmf-x86_64-sev\.bin/) {
+        record_info('OS Config Error', "Failed to properly update OS section. SEV-SNP firmware not configured.", result => 'fail');
+        return 0;
+    } else {
+        record_info('OS Config Success', "Successfully configured SEV-SNP compatible OS section");
+    }
+
+    # Add the SEV-SNP launch security section before </domain>
+    my $launch_security = <<EOL;
+<launchSecurity type="sev-snp">
+  <policy>0x00030000</policy>
+</launchSecurity>
+EOL
+    script_run("echo '$launch_security' > /tmp/launch_security.xml");
+    # Insert the launch security section right before the </domain> tag - using sed -e command
+    script_run("sed -i -e '/<\\/domain>/e cat /tmp/launch_security.xml' $xml_file");
+
```

Comment:
Would you please add a record_info to show the new xml file to re-define the vm? For easy debug.

Reply:
Done.